src/Elevator.java

```diff
@@ -0,0 +1,145 @@
+    }
+
+    private static void elevatorDelay(int seconds) {
+        try {
+            Thread.sleep(seconds * 1000);
+        } catch (InterruptedException e) {
+            System.err.format("IOException: %s%n", e);
+        }
+    }
+
+    private void openElevatorDoor(){
+        door = ElevatorDoor.OPEN;
+        elevatorDelay(DOOR_OPENING_CLOSING_DELAY);
+    }
+
+    private void closeElevatorDoor(){
+        door = ElevatorDoor.CLOSE;
+        elevatorDelay(DOOR_OPENING_CLOSING_DELAY);
+    }
+
+
+    public boolean moveDown(int destinationFloor){
+        if(door == ElevatorDoor.CLOSE){
+            if(motor == ElevatorMotor.STOP || motor == ElevatorMotor.DOWNWARD)
+            {
+                System.out.println("Elevator Going DOWN");
+                elevatorLevel -= 1;
+                elevatorFloorButtons.put(destinationFloor, ElevatorButton.ON);
+                elevatorArrivalIndicator.put(destinationFloor, ArrivalIndicator.REACHED_FLOOR);
+                elevatorDelay(ELEVATOR_MOVING_TIME);
+                motor = ElevatorMotor.DOWNWARD;
+                System.out.println("Elevator " + this.elevatorNumber + ": Is Currently At " + elevatorLevel +"Floor");
+                return true;
+            }else{
+                System.out.println("Elevator Not Moving In The Requested Direction");
+                System.out.println("Current Elevator Direction: "+ motor.toString());
+                floors.add(new floorSubSystem(destinationFloor));
+            }
+        }
+        motor = ElevatorMotor.STOP;
+        elevatorFloorButtons.put(destinationFloor, ElevatorButton.ON);
+        elevatorArrivalIndicator.put(destinationFloor, ArrivalIndicator.NOT_REACHED_FLOOR);
+        closeElevatorDoor();
+        return false;
+    }
+
+
+    public boolean moveUp(int destinationFloor){
+        if(door == ElevatorDoor.CLOSE) {
+            if(motor == ElevatorMotor.STOP || motor == ElevatorMotor.UPWARD){
+                System.out.println("Elevator Going UP");
+                elevatorLevel += 1;
+                elevatorFloorButtons.put(destinationFloor, ElevatorButton.ON);
+                elevatorArrivalIndicator.put(destinationFloor, ArrivalIndicator.REACHED_FLOOR);
+                elevatorDelay(ELEVATOR_MOVING_TIME);
+                motor = ElevatorMotor.UPWARD;
+                System.out.println("Elevator " + this.elevatorNumber + ": Is Currently At " + elevatorLevel +"Floor");
+                return true;
+            }else{
+                System.out.println("Elevator Not Moving In The Requested Direction");
+                System.out.println("Current Elevator Direction: "+ motor.toString());
+                floors.add(new floorSubSystem(destinationFloor));
+            }
+        }
+        motor = ElevatorMotor.STOP;
+        elevatorFloorButtons.put(destinationFloor, ElevatorButton.ON);
+        elevatorArrivalIndicator.put(destinationFloor, ArrivalIndicator.NOT_REACHED_FLOOR);
```

Comment:
init all values in array to off based off the the number of floors then update as we reach and turn off when we go off the floor